### ISSUE TITLE
Enable Chromium flag #windows11-mica-titlebar to address white flash issues on Windows

### ISSUE THREAD

**Issue by Void48** [electron/electron#44689]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Electron apps on Windows can show a "white flash" when restoring a minimized window. This disrupts the user experience, especially in apps with dark themes.
This issue often frustrates me a lot, and I believe many dark mode users feel the same. I think most people just get used to it since it usually only appears for a split second, and full-window white flashes are rare. It’s not terrible, but it would be amazing if it never appeared at all.

### Proposed Solution

I’m requesting support for the Chromium flag **#windows11-mica-titlebar** in Electron. This flag addresses a common issue on Windows where a "white flash" appears when restoring a minimized Electron window. Allowing this flag as an argument would greatly improve the user experience, especially for applications with dark themes. ;)

In short, I’d love for the "**--enable-features=windows11-mica-titlebar**" argument to actually be supported.

Please let me know if more info is needed, but testing minimizing/maximizing in Chromium with the flag enabled or disabled should be straightforward.

### Alternatives Considered

I tried switching to different ANGLE graphics backends like D3D9 and OpenGL. While they did get rid of the "white flash", the performance was much worse compared to the default ANGLE backend, making them an unrealistic solution for apps that should be fast and responsive.

### Additional Information

Also, technically this solution would only work on Windows 11, but I think it’s still much better than having no solution at all.
+ support for Windows 10 will end in just under 11 months.

**Comment by GhostlyDark** [comment]
The white flash alongside resizing performance issues have been introduced in E27. I believe a corresponding Chromium version introduced changes that caused these regressions, as they also happen upstream. E26 and earlier don't exhibit these issues, E27 and newer do - same as Chromium.

This also happens on Windows 10 and the workaround for this on Chromium is to switch the default ANGLE graphics backend. 

To work around this issue as well as for debugging purposes, I'd like to suggest to expose the command line argument for changing the default ANGLE graphics backend.

**Comment by cxl258** [comment]
How do you add this parameter?

**Comment by Vasdranna** [comment]
Are there any news regarding this issue? It's very, very annoying.

**Comment by SunsetTechuila** [comment]
~https://chromium-review.googlesource.com/c/chromium/src/+/4552330~

~https://issues.chromium.org/issues/40284682~

https://issues.chromium.org/issues/427499162#comment2

**Comment by Magnate1234** [comment]
> ### Preflight Checklist
> * [x]  I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.[x]  I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.[x]  I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.
> 
> ### Problem Description
> Electron apps on Windows can show a "white flash" when restoring a minimized window. This disrupts the user experience, especially in apps with dark themes. This issue often frustrates me a lot, and I believe many dark mode users feel the same. I think most people just get used to it since it usually only appears for a split second, and full-window white flashes are rare. It’s not terrible, but it would be amazing if it never appeared at all.
> 
> ### Proposed Solution
> I’m requesting support for the Chromium flag **#windows11-mica-titlebar** in Electron. This flag addresses a common issue on Windows where a "white flash" appears when restoring a minimized Electron window. Allowing this flag as an argument would greatly improve the user experience, especially for applications with dark themes. ;)
> 
> In short, I’d love for the "**--enable-features=windows11-mica-titlebar**" argument to actually be supported.
> 
> Please let me know if more info is needed, but testing minimizing/maximizing in Chromium with the flag enabled or disabled should be straightforward.
> 
> ### Alternatives Considered
> I tried switching to different ANGLE graphics backends like D3D9 and OpenGL. While they did get rid of the "white flash", the performance was much worse compared to the default ANGLE backend, making them an unrealistic solution for apps that should be fast and responsive.
> 
> ### Additional Information
> Also, technically this solution would only work on Windows 11, but I think it’s still much better than having no solution at all. + support for Windows 10 will end in just under 11 months.

[electron-main-1.zip](https://github.com/user-attachments/files/20968924/electron-main-1.zip)

**Comment by gpetrov** [comment]
@Magnate1234 please do not post any zip files. Those will not be opened due to security considerations.

If you want to share code open an PR. Do explain what is it for when posting.

**Comment by sdhjjkadhjahjsd** [comment]
I used app.disableHardwareAcceleration(); and that solved the problem for me.
